radis/lbl/broadening.py

```diff
@@ -889,7 +889,7 @@ def _check_accuracy(self, df, wstep):
 
         ERROR_TRESHOLD = 1
 
-        if self.params.wstep == "auto":
+        if self.wstep == "auto":
```

Comment:
I think their should be no special treatment for 'auto' method. By construction it should pass the tests, but that's not a reason not to test it (what if someone somehow ends up manually resetting the wstep value after it was set by 'auto' ? )

Reply:
Actually, this line is required to prevent `self.wstep = 'auto'`  to have an arithmetic comparison in the next line because till this point we won't have calculated value of `wstep`, so this line is necessary. So tests cases will pass but the `auto` method won't work.

Reply:
This test is run after all FWHM have been computed already, so it should be possible to move the 'auto'  evaluation right before the test, either move the accuracy test right after 'auto'  was evaluated.

Reply:
Earlier we were doing the `auto` evaluation in `_check_accuracy` itself. So I implemented the later suggestion. I broke the `_check_accuracy` step in 2 parts - `_calc_min_width` to get the minimum width and `_check_accuracy` to check accuracy of wstep. Both these functions are now being called in `_generate_wavenumber_arrays` function so all computation and checks will be done in a single function.

Reply:
Yes, it's a much better architecture with the two functions ; well done !